tests/models/bark/test_modeling_bark.py

```diff
@@ -523,21 +617,166 @@ def get_pipeline_config(self):
         return config
 
     def prepare_config_and_inputs_for_common(self):
-        # TODO: @Yoach
-        pass
-        # return config, inputs_dict
+        config, inputs_dict = self.prepare_config_and_inputs()
+        return config, inputs_dict
 
 
 # Need this class in oder to create tiny model for `bark`
-# TODO (@Yoach) Implement actual test methods
-@unittest.skip("So far all tests will fail.")
-class BarkModelTest(ModelTesterMixin, GenerationTesterMixin, unittest.TestCase):
-    all_model_classes = (BarkModel,) if is_torch_available() else ()
+class BarkModelTest(unittest.TestCase):
```

Comment:
Without this being a subclass of `ModelTesterMixin`, there is no way to create tiny model for `Bark`. 

As you mentioned there is even no `forward` for `BarkModel`, subclassing and overwritting seems too much of work for nothing.

Let's see what @amyeroberts and @sgugger would say, but I could accept this and try to modify the tiny model script to make it work for `bark`.

(the `run_pipeline_test` for pipeline testing would not be run if `PipelineTesterMixin` is not included here. But this is not the point for this comment)

Reply:
It's a slippery slope though. It's a sign that the model does not fit the API of Transformers, so what is even the goal of having it in the library in that case? Transformers strength is to provide a common API for all models after all.

Reply:
Just so I understand better - what's the motivation behind creating a tiny version of the model? Is it used in independently to the tests in [`test_modeling_bark.py`](https://github.com/huggingface/transformers/blob/main/tests/models/bark/test_modeling_bark.py), where a config is defined explicitly and a model created based on these values?

Bark is irregular in the sense that it's a concatenation of three auto-regressive models, meaning there's no notion of a forward pass (see [comment](https://github.com/huggingface/transformers/pull/24086#issuecomment-1620555118) for details). There was a long discussion about this on the PR, where we agreed that having a `.generate` method was the best fit for the Transformers API, and having a `.forward` method doesn't really work here.

I think @ylacombe has very nicely described the motivations for including the model in the Transformers library, but it's a very valid point from @sgugger that overall it's not the best fit. Maybe in the long term we can start a new TTS library that is more flexible for these kinds of audio architectures (given we're on the limit of the Transformers design), or perform a Transformers refactor to better facilitate these kinds of new TTS models

Reply:
> what's the motivation behind creating a tiny version of the model

The motivation is to be able to run pipeline testing, for example, the one added by @ylacombe in

https://github.com/huggingface/transformers/blob/7e141278cab9121d8d036db103db2edbeaab2ce8/tests/pipelines/test_pipelines_text_to_audio.py#L228

which is linked to [PipelineTesterMixin](https://github.com/huggingface/transformers/blob/7e141278cab9121d8d036db103db2edbeaab2ce8/tests/test_pipeline_mixin.py).

Since 10 month ago, we decided to move the logic of their creation outside the testing. But even if they are inside the testing code, the creation would fail for `bark`.

> a config is defined explicitly and a model created based on these values?

This is true, but they are not for the pipeline testing, only the usual model tests.

Reply:
Do we really need to run any test outside of a slow integration test?

Reply:
@sgugger 

>Do we really need to run any test outside of a slow integration test?

If you mean the slow integration tests implemented in each model test file: yes, we need to test the model when it is used as a pipeline.


For each pipeline type, we have 2 kind of tests:

  - the ones directly runs in `tests/pipelines/test_pipelines_xxx.py` (like `test_small_model_pt` etc)
    - not related to tiny model stuff
  - the `run_pipeline_test` which is coupled with model test file via `PipelineTesterMixin`
    - involved tiny model

We have `run_pipeline_test` for each pipeline task (didn't check if there is any exception) for relevant model classes. It's the way for us to know if a model fits a pipeline tasks, but @Narsil might have better comment here.

However, [here](https://github.com/huggingface/transformers/blob/7e141278cab9121d8d036db103db2edbeaab2ce8/tests/pipelines/test_pipelines_text_to_audio.py) already has some pipeline tests using `bark`, so we could say it's fine not to have a tiny bark running against `run_pipeline_test`.